pkg/build/builder/docker.go

```diff
@@ -416,6 +426,49 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV instruction with the environment variables
+// from env after every FROM instruction in the given Dockerfile instructions
+// and returns a new Dockerfile.
+func insertEnvAfterFrom(instructions []byte, env []kapi.EnvVar) ([]byte, error) {
+	if len(env) == 0 {
+		return instructions, nil
+	}
+	node, err := parser.Parse(bytes.NewReader(instructions))
+	if err != nil {
+		return nil, err
+	}
+
+	// buildEnv is an ENV Dockerfile instruction to define all environment
+	// variables from env.
+	buildEnv := []byte("ENV")
```

Comment:
I disagree. "ENV" is easier to read here than "strings.ToUpper(dockercmd.Env)". And BTW it doesn't have to be in uppercase as the Dockerfile format is case insensitive.

The idea of the abstraction here is that you can insert "a Dockerfile" into another "Dockerfile" at any given position. Would you write a Dockerfile as:

```
LABEL version=1.0
ENV PATH=/bin/bash
```

Or would you interpolate every command to come from a constant defined somewhere?